Modules/+Drivers/Keithley2400.m

```diff
@@ -0,0 +1,148 @@
+            end
+            for i = 1:length(Objects)
+                if isvalid(Objects(i)) && isequal(GPIBAddr,Objects(i).singleton_id)
+                    obj = Objects(i);
+                    return
+                end
+            end
+            obj = Drivers.Keithley2400(GPIBAddr);
+            obj.singleton_id = GPIBAddr;
+            Objects(end+1) = obj;
+        end
+    end
+    methods (Access=private)
+        function obj = Keithley2400(GPIBAddr)
+            obj.GPIBAddr = GPIBAddr;
+            obj.RsrcName = ['GPIB0::' num2str(obj.GPIBAddr) '::INSTR'];
+            obj.VisaHandle = visa('ni',obj.RsrcName);
+            obj.openConnection();
+        end
+    end
+    
+    % I/O functions
+    methods
+        function openConnection(obj)
+            if strcmp(obj.VisaHandle.Status,'closed')
+                fopen(obj.VisaHandle);
+                obj.init();     % Read values from the source.
+            end
+        end
+        function closeConnection(obj)
+            if strcmp(obj.VisaHandle.Status,'open')
+                fclose(obj.VisaHandle);
+            end
+        end
+        function delete(obj)
+            obj.closeConnection();
+            delete(obj.VisaHandle);
+        end
+        function cmd(obj,msg)
+            % open the port if closed
+            if strcmp(obj.VisaHandle.Status,'closed')
+                fopen(obj.VisaHandle);
+                CloseOnDone = 1;
+            else
+                CloseOnDone = 0;
+            end
+            fprintf(obj.VisaHandle,msg);
+            if CloseOnDone
+                fclose(obj.VisaHandle);
+            end
+        end
+        function out = query(obj,msg)
+            % open the port if closed
+            if strcmp(obj.VisaHandle.Status,'closed')
+                fopen(obj.VisaHandle);
+                CloseOnDone = 1;
+            else
+                CloseOnDone = 0;
+            end
+            fprintf(obj.VisaHandle,msg);
+            out = fscanf(obj.VisaHandle);
+            if CloseOnDone
+                fclose(obj.VisaHandle);
+            end
+        end
+        function init(obj)
+            obj.output =    obj.get_output();
+%             obj.mode =      obj.get_mode();
+            obj.voltage =   obj.get_voltage();
+            obj.current =   obj.get_current();
+        end
+    end
+    
+    % Basic control functions
+    methods
+        % Output
+        function val = set_output(obj, val, ~)
+            val = logical(val);
+            obj.cmd(['outp:stat ' num2str(val)]);
+        end
+        function output = get_output(obj)
+            output = str2double(obj.query('outp:stat?'));
+            obj.output = output;
+        end
+        
+        % Mode
+        function val = set_mode(obj, val, ~)
+            assert(strcmp(val, 'VOLT') || strcmp(val, 'CURR'), 'mode must be: ''VOLT'' or ''CURR''');
+            obj.cmd(['sour:func:mode ' val]);
+        end
+%         function mode = get_mode(obj)
```

Comment:
Should this be implemented, or removed?

Reply:
It should be implemented, but it isn't a priority now.